mobile/src/main/java/org/openhab/habdroid/ui/WidgetAdapter.kt

```diff
@@ -557,6 +584,185 @@ class WidgetAdapter(
         }
     }
 
+    class InputViewHolder internal constructor(initData: ViewHolderInitData) :
+        LabeledItemBaseViewHolder(initData, R.layout.widgetlist_inputitem, R.layout.widgetlist_inputitem_compact) {
+        private val inputTextLayout: TextInputLayout = itemView.findViewById(R.id.widgetinput)
+        private val inputText: TextInputEditText = itemView.findViewById(R.id.widgetinputvalue)
+        private var isBinding = false
+        private var hasChanged = false
+        private var showingDatePicker = false
+        private var showingTimePicker = false
+
+        private var updateJob: Job? = null
+        private var oldValue: String? = null
+
+        init {
+            inputText.doAfterTextChanged {
+                if (!isBinding) {
+                    hasChanged = true
+                }
+            }
+            inputText.setOnFocusChangeListener { _, hasFocus ->
+                if (!hasFocus && hasChanged) {
+                    updateValue()
+                }
+            }
+            inputText.setOnClickListener {
```

Comment:
Thinking about it, I think it would make more sense to instead override `handleRowClick()` and there to
- open date or time picker if appropriate
- if no picker is needed, focus the input field (`inputText.requestFocus()`)

Reply:
Done, I still needed to inputText.setOnClickListener to pass on the handleRowClick. If not, it only worked when clicking outside of the text field.